projects/loopFinance/index.js

```diff
@@ -0,0 +1,104 @@
+const sdk = require("@defillama/sdk");
+const { fetchURL } = require('../helper/utils')
+const { getBalance, getDenomBalance } = require("../helper/terra");
+const { getBlock } = require("../helper/getBlock");
+
+async function getPairs() {
+    const factory00 = (await fetchURL(
+        queries.baseUrl + queries.factory0 + queries.query30
+    )).data.result.pairs;
+    const factory01 = (await fetchURL(
+        queries.baseUrl + queries.factory0 + queries.factory0query60
+    )).data.result.pairs;
+    const factory10 = (await fetchURL(
+        queries.baseUrl + queries.factory1 + queries.query30
+    )).data.result.pairs;
+
+    if (factory01.length > 29 || factory10.length > 29) {
+        throw new Error(`We need another query since they're capped at 30 results and pools may be missing`);
+    };
+
+    return factory00.concat(factory01, factory10);
+};
+
+const tokenMapping = {
```

Comment:
at some point this can be merged with `helper/terra.js`, there is a mapping present that. Maybe later, I can move both to `portedTokens.js`